fe/src/main/java/org/apache/doris/alter/Alter.java

```diff
@@ -273,6 +276,18 @@ public void processAlterTable(AlterTableStmt stmt) throws UserException {
                 throw new DdlException("table with empty parition cannot do schema change. [" + tableName + "]");
             }
 
+            // if table state is unhealthy, change table repair priority, and wait until repair finish or exceed timeout
+            if (olapTable.getState() != OlapTableState.NORMAL) {
```

Comment:
First, `olapTable.getState() != OlapTableState.NORMAL` does not mean the table is unhealthy.
Second, it's not good to sleep here, it will block the client connection for a very long time.
Third, I don't think its good idea to add priority repair here.  We should solve more why the table is often unhealthy, rather than increase the complexity here.

Reply:
@morningman Yes, you're right, I change the waiting logic to runPendingJob.